contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -7681,6 +7944,266 @@ rewrite_column_name_with_omitted_schema_name(T ctx, GetCtxFunc<T> getSchema, Get
+	keysToRemove.clear();
+	
+	/* shifting the local id positions to new positions after rewriting the query since they will be quoted later */
+	for (auto &entry : local_id_positions)
+	{
+		if(entry.first >= ctx->start->getStartIndex() && entry.first <= geospatial_start_index)
+		{
+			keysToRemove.push_back(entry.first);
+			local_id_positions.emplace(std::make_pair(entry.first + method->spatial_methods()->expression_list()->stop->getStopIndex() - method->spatial_methods()->geospatial_func_arg()->start->getStartIndex() + 2, entry.second));
+		}
+		else if(entry.first >= method->spatial_methods()->expression_list()->start->getStartIndex() && entry.first <= method->spatial_methods()->expression_list()->stop->getStopIndex())
+		{
+			size_t pos = entry.first;
+			size_t offset = method->start->getStartIndex() - ctx->start->getStartIndex();
+			pos -= offset;
+			keysToRemove.push_back(entry.first);
+			local_id_positions.emplace(std::make_pair(pos, entry.second));
+		}
+	}
+	for (const auto &key : keysToRemove) local_id_positions.erase(key);
+	keysToRemove.clear();
+	expr += ctx_str.substr(index);
+	std::string rewritten_exp = expr.substr((int)method->start->getStartIndex() - ctx->start->getStartIndex() + offset1, method_len + offset2) + "," + expr.substr(0, func_call_len + offset1 + 1) + ")";
+	if ((int)method->stop->getStopIndex() - ctx->start->getStartIndex() + 1 < ctx_str.size()) rewritten_exp += expr.substr(method->stop->getStopIndex() + offset1 - ctx->start->getStartIndex() + 1);
+	rewritten_query_fragment.emplace(std::make_pair(ctx->start->getStartIndex(), std::make_pair(ctx_str.c_str(), rewritten_exp.c_str())));
+}
+
+/*
+ * In this helper function we Rewrite the Query for Geospatial Handling
+ * For Col_Ref Functions (such as STX, STY, Lat, Long) : ColRef.Func_name  ->  (ColRef).Func_name
+ * For Func_Ref Functions (such as STAsText, STAsBinary, STDistance) : ColRef.Func_name (arg_list)  ->  Func_name (arg_list, ColRef)
+ */
+template<class T>
+void
+rewrite_geospatial_query_helper(T ctx, TSqlParser::Method_callContext *method, size_t geospatial_start_index)
+{	
+	/* Check whether it is a Col_Ref Function or a Func_Ref Function */
+	if(method->spatial_methods()->geospatial_col() && !method->spatial_methods()->LR_BRACKET() ) 
+		rewrite_geospatial_col_ref_query_helper(ctx, method, geospatial_start_index);
+	else if(method->spatial_methods()->geospatial_func_no_arg() && !method->spatial_methods()->expression_list() ) 
+		rewrite_geospatial_func_ref_no_arg_query_helper(ctx, method, geospatial_start_index);
+	else if(method->spatial_methods()->geospatial_func_arg() && method->spatial_methods()->expression_list() ) 
+		rewrite_geospatial_func_ref_args_query_helper(ctx, method, geospatial_start_index);
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// Spatial Query Helper for Function Calls
+////////////////////////////////////////////////////////////////////////////////
```

Comment:
Nit: Should have corresponding closing block at the end.

Reply:
Done